src/main/java/com/github/mauricioaniche/ck/metric/TightClassCohesion.java

```diff
@@ -0,0 +1,128 @@
+package com.github.mauricioaniche.ck.metric;
+
+import com.github.mauricioaniche.ck.CKClassResult;
+import com.github.mauricioaniche.ck.CKMethodResult;
+import com.google.common.collect.Sets;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+//Calculates the tight and loose class cohesion for a class.
+//For more details see: https://www.aivosto.com/project/help/pm-oo-cohesion.html#TCC_LCC
+public class TightClassCohesion implements CKASTVisitor, ClassLevelMetric {
+    private HashMap<String, Set<String>> accessedFields = new HashMap<>();
+
+    //Two methods are directly connected if:
```

Comment:
what's a _class level variable_? A _field_?

Reply:
That's my understanding from the definition,